src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -186,7 +186,7 @@ public void Dispose()
             }
         }
 
-        internal object CaptureDisposable(object service)
+        internal virtual object CaptureDisposable(object service)
```

Comment:
> I could be convinced of the value of the test, but arguing it's more valuable then real benchmarks that are of course monitored for regressions is laughable. I understand we live in the real world though, and what I'm asking for could be a long time coming.

@pakrym is going to add some benchmark.net tests to this repo.

> Do we know if expression tree compilation will ever inline this method? Could that be changed by visualizing it? Even if CaptureDisposable was never inlined in the first place, doesn't mean making this virtual is free.

I'm going to change it from being virtual so we don't even theoretically affect the performance. Will add an internal callback instead.